tv_apps/dashboard/test/unit/digital_clock_test.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+/* global DigitalClock */
+
+require('/js/digital_clock.js');
+
+suite('DigitalClock', function() {
+  var clock;
+  var digitalClock;
+  var startDate;
+
+  function createElement(tagName, id) {
+    var element = document.createElement(tagName);
+    element.id = id;
+
+    return element;
+  }
+
+  suiteSetup(function() {
+    document.body.appendChild(createElement('span', 'hours-tens-digit'));
+    document.body.appendChild(createElement('span', 'hours-units-digit'));
+    document.body.appendChild(createElement('span', 'minutes-tens-digit'));
+    document.body.appendChild(createElement('span', 'minutes-units-digit'));
+    document.body.appendChild(createElement('span', 'seconds-tens-digit'));
+    document.body.appendChild(createElement('span', 'seconds-units-digit'));
+    document.body.appendChild(createElement('span', 'weekday'));
+    document.body.appendChild(createElement('span', 'month'));
+    document.body.appendChild(createElement('span', 'date'));
+  });
+
+  suiteTeardown(function() {
+    var elements = document.getElementsByTagName('span');
+    for (var i=0; i < elements.length; i++) {
+      document.body.removeChild(elements[i]);
+    }
+  });
+
+  setup(function() {
+    startDate = new Date(2015, 4, 31);
+    clock = sinon.useFakeTimers(startDate.getTime());
+    digitalClock = new DigitalClock();
+    digitalClock.init();
+  });
+
+  teardown(function() {
+    digitalClock.stop();
+    clock.restore();
+  });
+
+  test('Should display current date and time after _initTime()', function() {
+    assert.equal(digitalClock.date.getTime(), startDate.getTime());
+    assert.equal(digitalClock.elements['hours-tens-digit'].textContent, '0');
+    assert.equal(digitalClock.elements['hours-units-digit'].textContent, '0');
+    assert.equal(digitalClock.elements['minutes-tens-digit'].textContent, '0');
+    assert.equal(digitalClock.elements['minutes-units-digit'].textContent, '0');
+    assert.equal(digitalClock.elements['seconds-tens-digit'].textContent, '0');
+    assert.equal(digitalClock.elements['seconds-units-digit'].textContent, '0');
+    assert.equal(digitalClock.elements.weekday.dataset.l10nId,
+                 'weekday-0-long');
+    assert.equal(digitalClock.elements.month.dataset.l10nId, 'month-4-short');
+    assert.equal(digitalClock.elements.date.textContent, '31');
+  });
+
+  test('Should run the passed in callback every second after _start()',
+    function() {
+      var callback = sinon.spy();
+      digitalClock._start(callback);
```

Comment:
Actually I think this test case is not required and you can put the ticking to the case `Should update date and time after _update()`.